dds/DCPS/DataReaderImpl_T.h

```diff
@@ -954,16 +954,25 @@ namespace OpenDDS {
       ser.encoding(encoding);
     }
 
+    bool ser_ret = true;
     MessageType data;
     if (sample.header_.key_fields_only_) {
-      ser >> OpenDDS::DCPS::KeyOnly<MessageType>(data);
+      ser_ret = ser >> OpenDDS::DCPS::KeyOnly<MessageType>(data);
     } else {
-      ser >> data;
-    }
-    if (!ser.good_bit()) {
-      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) %CDataReaderImpl::lookup_instance ")
-                 ACE_TEXT("deserialization failed.\n"),
-                 TraitsType::type_name()));
+      ser_ret = ser >> data;
+    }
+    if (!ser_ret) {
+      if (ser.get_construction_status() != Serializer::ConstructionSuccessful) {
+        if (DCPS_debug_level > 1) {
+          ACE_DEBUG((LM_WARNING, ACE_TEXT("(%P|%t) %CDataReaderImpl::lookup_instance ")
+                    ACE_TEXT("object construction failure, dropping sample.\n"),
+                    TraitsType::type_name()));
```

Comment:
```suggestion
                     ACE_TEXT("object construction failure, dropping sample.\n"),
                     TraitsType::type_name()));
```

Same for the other new logs.